### ISSUE TITLE
text to speech not working

### ISSUE THREAD

**Issue by hpavenge** [Microsoft/BotFramework-Emulator#412]
simply adding speak to the call doesnt generate text to speech in the emulator? following this tutorial it should: https://docs.microsoft.com/en-us/bot-framework/dotnet/bot-builder-dotnet-text-to-speech#a-idssmla-speech-synthesis-markup-language-ssml

what am i missing?

**Comment by EricDahlvang** [comment]
Hi @hpavenge 

That doc should state: "To use speech in the emulator, the bot must be registered and using a valid MS AppId and Password."  ref: https://github.com/Microsoft/BotFramework-Emulator/blob/c014d0494cd4eaefa5e1b7d1fd6e745c1c2ad504/src/server/conversationManager.ts#L461

Thank you for bringing this to our attention.

**Comment by hpavenge** [comment]
They are registered with a valid ms appid and password. Also tried hosting it with ngrok. Still no sound out of the emulator while the speak message is filled in the details json. FYI I'm using the Bot framework emulator as a desktop cliënt.

**Comment by hpavenge** [comment]
Okay here goes for anyone having the same problem. I was just sending text messages and expecting the bot to speak back. It doesn't speak back untill u actually use the mic. I wasn't using the mic because the bing speech api doesn't understand dutch. 

**TLDR** use the mic in the emulator. Doesnt work with typing.

**Comment by gouravbeyondBGCM** [comment]
@EricDahlvang you said AppId and password has to be in Bots settings , where is that setting , and how can I specify that .

**Comment by EricDahlvang** [comment]
in a .net Bot, you can specify the MicrosoftAppId and MicrosoftAppPassword in the web.config as settings:
  ```xml
  <add key="MicrosoftAppId" value="YourMicrosoftAppId" />
    <add key="MicrosoftAppPassword" value="YourMicrosoftAppPassword" />
  ```

For a node bot, if you are using Visual Studio Code, you can use the 'dotenv-extended' package, and provide a .env file that contains the appid and password (here is an example: https://github.com/Microsoft/BotBuilder-Samples/blob/master/Node/core-DirectLineWebSockets/DirectLineBot/.env )

Then, supply the same appid and password in the emulator:

![image](https://user-images.githubusercontent.com/11055362/40560547-327db9c4-600f-11e8-88c1-87e0c598df4f.png)

AppId and Password are required for speech to work correctly in the emulator.
